src/test/ui/const-generics/projection-as-arg-const.rs

```diff
@@ -0,0 +1,17 @@
+// run-pass
+
+pub trait Identity {
+    type Identity;
+}
+
+impl<T> Identity for T {
+    type Identity = Self;
+}
+
+pub fn foo<const X: <i32 as Identity>::Identity>() {
```

Comment:
tbh I am not sure whether i want to insta stabilize projections in const params or keep them as part of the `adt_const_params` feature :thinking: needs a lang FCP if insta stable

Reply:
As you prefer! If a FCP is needed, I'll let someone from your team handle this part.